lib/services/network/http_service.dart

```diff
@@ -1,14 +1,224 @@
 import 'package:bluebubbles/models/models.dart';
+import 'package:bluebubbles/utils/file_utils.dart';
 import 'package:bluebubbles/utils/logger.dart';
 import 'package:bluebubbles/services/services.dart';
 import 'package:dio/dio.dart';
 import 'package:flutter/foundation.dart';
 import 'package:flutter_test/flutter_test.dart';
 import 'package:get/get.dart' hide Response, FormData, MultipartFile;
+import 'backend_service.dart';
 
 /// Get an instance of our [HttpService]
 HttpService http = Get.isRegistered<HttpService>() ? Get.find<HttpService>() : Get.put(HttpService());
 
+class HttpBackend implements BackendService {
+  @override
+  Future<Chat> createChat(List<String> addresses, String? message, String service, {CancelToken? cancelToken}) async {
+    var response = await http.createChat(addresses, message, service, cancelToken: cancelToken);
+    return Chat.fromMap(response.data["data"]);
+  }
+
+  @override
+  void init() { }
+
+  @override
+  void startedTyping(Chat c) {
+    socket.sendMessage("started-typing", {"chatGuid": c.guid});
+  }
+
+  @override
+  void stoppedTyping(Chat c){
+    socket.sendMessage("stopped-typing", {"chatGuid": c.guid});
+  }
+
+  @override
+  void updateTypingStatus(Chat c) {
+    socket.sendMessage("update-typing-status", {"chatGuid": c.guid});
+  }
+  
+  @override
+  Future<Message> sendMessage(Chat c, Message m, {CancelToken? cancelToken}) async {
+    var response = await http.sendMessage(c.guid,
+        m.guid!,
+        m.text!,
+        subject: m.subject,
+        method: (ss.settings.enablePrivateAPI.value
+            && ss.settings.privateAPISend.value)
+            || (m.subject?.isNotEmpty ?? false)
+            || m.threadOriginatorGuid != null
+            || m.expressiveSendStyleId != null
+            ? "private-api" : "apple-script",
+        selectedMessageGuid: m.threadOriginatorGuid,
+        effectId: m.expressiveSendStyleId,
+        partIndex: int.tryParse(m.threadOriginatorPart?.split(":").firstOrNull ?? ""),
+        ddScan: m.text!.isURL, cancelToken: cancelToken);
+    return Message.fromMap(response.data["data"]);
+  }
+  
+  @override
+  Future<bool> renameChat(Chat chat, String newName) async {
+    return (await http.updateChat(chat.guid, newName)).statusCode == 200;
```

Comment:
This is assumed to always be true because the future only completes successfully with code 200